lib/remap-functions/src/length.rs

```diff
@@ -0,0 +1,136 @@
+use remap::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct Length;
+
+impl Function for Length {
+    fn identifier(&self) -> &'static str {
+        "length"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            accepts: |v| matches!(v, Value::Array(_) | Value::Map(_)),
+            required: true,
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+
+        Ok(Box::new(LengthFn { value }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct LengthFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for LengthFn {
+    fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
+        use Value::*;
+
+        let value = self.value.execute(state, object)?;
+
+        match value {
+            Array(v) => Ok((v.len() as i32).into()),
```

Comment:
Nit: these will be turned into `i64`s so I might just do the cast to i64 here.

There is a potential for silent overflow with a vec or map that has > the max i64 elements (given it uses `usize`) but the risk of people having such large events seems pretty small. We could do a safe cast if we wanted via:

```rust
i64::try_from(v.len()).into()
```

And returning an error here if there was overflow.

Reply:
Better safe than sorry. Good suggestion.